R/lazy-join-query.R

```diff
@@ -65,7 +65,7 @@ join_check_vars <- function(vars, call) {
 }
 
 join_check_by <- function(by, call) {
-  if (!vctrs::vec_is_list(by)) {
+  if (!vctrs::vec_is_list(by) && !inherits(by, "dplyr_join_by")) {
     # TODO use `cli_abort()` after https://github.com/r-lib/rlang/issues/1386
     # is fixed
     abort("`by` must be a list", .internal = TRUE)
```

Comment:
I think since you're touching the line above you could switch to `cli_abort()` since the bug is fixed.